.circleci/config.yml

```diff
@@ -0,0 +1,240 @@
+          command: |
+            sudo sysctl -w vm.max_map_count=262144
+            sudo sysctl vm.max_map_count
+      # Run back-end (Django) check
+      #
+      # Nota bene: to run the django app, we need to ensure that both PostgreSQL
+      # and ElasticSearch services are up and ready. To achieve this, we wrap
+      # commands execution with dockerize, a tiny tool installed in the CI
+      # image. In our case, dockerize will wait up to one minute that both the
+      # database and elastisearch containers opened their expected tcp port
+      # (5432 and 9200 resp.).
+      - run:
+          name: Run Django check in production image
+          command: |
+            make run
+            sleep 10
+            make migrate
+            make check
+
+  # ---- DockerHub publication job ----
+  hub:
+    machine: true
+    working_directory: ~/fun
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - docker-images-ci-{{ .Revision }}
+      # Load all built images in all flavors
+      - run:
+          name: Load images to docker engine
+          command: |
+            docker load < docker/images/funmooc.tar
+      # Login to DockerHub to Publish new images
+      #
+      # Nota bene: you'll need to define the following secrets environment vars
+      # in CircleCI interface:
+      #
+      #   - DOCKER_USER
+      #   - DOCKER_PASS
+      - run:
+          name: Login to DockerHub
+          command: echo "$DOCKER_PASS" | docker login -u "$DOCKER_USER" --password-stdin
+      # Tag docker images with the same pattern used in Git (Semantic Versioning)
+      #
+      # Git tag: v1.0.1
+      # Docker tag: 1.0.1(-alpine)(-ci)
+      - run:
+          name: Tag images
+          command: |
+            docker images fundocker/funmooc
+            DOCKER_TAG=$([[ -z "$CIRCLE_TAG" ]] && echo $CIRCLE_BRANCH || echo ${CIRCLE_TAG} | sed 's/^v//')
+            RELEASE_TYPE=$([[ -z "$CIRCLE_TAG" ]] && echo "branch" || echo "tag ")
+            # Display either:
+            # - DOCKER_TAG: master (Git branch)
+            # or
+            # - DOCKER_TAG: 1.1.2 (Git tag v1.1.2)
+            echo "DOCKER_TAG: ${DOCKER_TAG} (Git ${RELEASE_TYPE}${CIRCLE_TAG})"
+            docker tag funmooc:${CIRCLE_SHA1}production fundocker/funmooc:${DOCKER_TAG}
+            if [[ -n "$CIRCLE_TAG" ]]; then
+                docker tag funmooc:${CIRCLE_SHA1}production fundocker/funmooc:latest
+            fi
+            docker images | grep -E "^fundocker/funmooc\s*(${DOCKER_TAG}.*|latest|master)"
+      # Publish images to DockerHub
+      #
+      # Nota bene: logged user (see "Login to DockerHub" step) must have write
+      # permission for the project's repository; this also implies that the
+      # DockerHub repository already exists.
+      - run:
+          name: Publish images
+          command: |
+            DOCKER_TAG=$([[ -z "$CIRCLE_TAG" ]] && echo $CIRCLE_BRANCH || echo ${CIRCLE_TAG} | sed 's/^v//')
```

Comment:
Did you try to put this in the task's `environment` section ? If it's possible to compute the value in one place I think we should do it.

Reply:
I've defined environment variables. Let's see how it goes.